### PR TITLE
Add 'remote_parameters' and 'get_module_rref' to RemoteModule docs.

### PR DESCRIPTION
**Pull request by pritamdamania87**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54645 Add 'remote_parameters' and 'get_module_rref' to RemoteModule docs.**

Had to replace RRef[..] with just RRef in the return signature since
sphynx seemed to completely mess up rendering RRef[..]

Screen Shot 2021-03-24 at 5 42 34 PM

Differential Revision: [D27314609](https://our.internmc.facebook.com/intern/diff/D27314609/)

### CODE REVIEW

File: torch/distributed/nn/api/remote_module.py [hunk 1]
```diff
@@ -196,9 +196,11 @@ def __init__(
             method = torch.jit.export(method)
             setattr(self, method_name, types.MethodType(method, self))
 
-    def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
-        r"""Returns a list of RRefs of remote module parameters.
+    def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef]:
+        """
+        Returns a list of RRefs of remote module parameters.
```
Comment (SciPioneer) [inline torch/distributed/nn/api/remote_module.py:201]: Since we have to change `RRef[Parameter]` to `RRef` in the signature, probably we can add `Parameter` back in the comment to make up this information loss?

For example:
r"""Returns a list of RRefs of remote module parameters (`List[rpc.RRef[Parameter]]`).

File: torch/distributed/nn/api/remote_module.py [hunk 2]
```diff
@@ -209,7 +211,7 @@ def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
         """
         return rpc.rpc_sync(self.on, _param_rrefs, args=(self.module_rref, recurse))
 
-    def get_module_rref(self) -> rpc.RRef[nn.Module]:
+    def get_module_rref(self) -> rpc.RRef:
```
Comment (SciPioneer) [inline torch/distributed/nn/api/remote_module.py:214]: Ditto

**Review by wayi1 — APPROVED** [review]
